scripts/utils.ts

```diff
@@ -1,48 +1,16 @@
 
 import { BigNumber } from 'ethers';
 import { decodeAddress } from '@polkadot/util-crypto';
-import { ethers, network } from 'hardhat';
 import { formatUnits, parseEther } from 'ethers/lib/utils';
 
-import { ADDRESSES } from './consts';
-
-export const loadSetups = async () => {
-  const [[deployer, user, relayer], FeeRegistry, Factory, Token] = await Promise.all([
-    ethers.getSigners(),
-    ethers.getContractFactory('FeeRegistry'),
-    ethers.getContractFactory('Factory'),
-    ethers.getContractFactory('MockToken'),
-  ]);
-
-  const { usdcAddr, factoryAddr, feeAddr } = ADDRESSES[network.name];
-
-  const usdt = Token.attach(usdcAddr);
-  const fee = FeeRegistry.attach(feeAddr);
-  const factory = Factory.attach(factoryAddr).connect(relayer);
-
-  console.log('setup finished');
-  console.log({
-    deployerAddr: deployer.address,
-    userAddr: user.address,
-    relayerAddr: relayer.address,
-    factoryAddr: factory.address,
-    usdcAddr: usdcAddr,
-    feeRegistryAddr: fee.address,
-    routerFee: ethers.utils.formatEther(await fee.getFee(usdcAddr)),
-  });
-  console.log('');
-
-  return { deployer, user, relayer, usdt, fee, factory, ...ADDRESSES[network.name] };
-};
+export type Resolved<T> = T extends Promise<infer U> ? U : T;
```

Comment:
There is already `Awaited` https://www.typescriptlang.org/docs/handbook/utility-types.html#example

Reply:
great to know, obviously I asked old chatgpt who doesn't know this yet lol

Reply:
I use phind instead of chatgpt https://www.phind.com/search?cache=shuz8urtzjp2lhxuibfhkgew

Reply:
oh this is really cool, will try it out! It even has a vscode extension